odo/backends/csv.py

```diff
@@ -330,6 +337,9 @@ def _():
 
 @discover.register(CSV)
 def discover_csv(c, nrows=1000, **kwargs):
+    if c._dshape:
+        return c._dshape
```

Comment:
Perhaps we could add an `ensure_consistent_dshape` default arg (default `False`); if set to `True`, then the `c._dshape` would be tested against the dshape of the `df` loaded below to ensure the user-specified dshape is compatible with the discovered dshape.